[LBSE] Fix use-on-clip-path-with-transformation.svg

rwlbuis opened the pull request:
#### 3f1e7f15120b48bfc0aa895a6783c5c1dfcf4f9a
<pre>
[LBSE] Fix use-on-clip-path-with-transformation.svg
<a href="https://bugs.webkit.org/show_bug.cgi?id=268283">https://bugs.webkit.org/show_bug.cgi?id=268283</a>

Reviewed by Nikolas Zimmermann.

pathFromGraphicsElement and the layer transform both contain the use x/y translate, so avoid
calling pathFromGraphicsElement in the use case by instead directly calling pathFromGraphicsElement
on the clip child.

* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::computeClipPath const):

Canonical link: <a href="https://commits.webkit.org/274074@main">https://commits.webkit.org/274074@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/be3081801cbcbf3c1fcdf85a93a1565ec3086a6f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37797 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16694 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40038 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40337 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33623 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/39124 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19347 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13956 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31978 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38364 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14056 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33097 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12281 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12217 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33789 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41597 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34230 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34233 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38104 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12805 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10412 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36278 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14218 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8489 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13185 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13528 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
```diff
@@ -290,6 +290,8 @@ Path RenderSVGModelObject::computeClipPath(AffineTransform& transform) const
     if (auto* useElement = dynamicDowncast<SVGUseElement>(element())) {
         if (auto* clipChild = useElement->rendererClipChild())
```
Maybe rename to clipChildRenderer to avoid confusion?

nikolaszimmermann reviewed (approved):
r=me with one comment from last review still open.

rwlbuis committed:
[LBSE] Fix use-on-clip-path-with-transformation.svg
https://bugs.webkit.org/show_bug.cgi?id=268283

Reviewed by Nikolas Zimmermann.

pathFromGraphicsElement and the layer transform both contain the use x/y translate, so avoid
calling pathFromGraphicsElement in the use case by instead directly calling pathFromGraphicsElement
on the clip child.

* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::computeClipPath const):

Canonical link: https://commits.webkit.org/274074@main
